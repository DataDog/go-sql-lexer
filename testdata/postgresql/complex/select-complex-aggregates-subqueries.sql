SELECT 
  u.id,
  u.name,
  (SELECT COUNT(*) FROM orders o WHERE o.user_id = u.id) AS order_count,
  (SELECT SUM(amount) FROM payments p WHERE p.user_id = u.id) AS total_payments,
  (SELECT AVG(rating) FROM reviews r WHERE r.user_id = u.id) AS average_rating
FROM 
  users u
WHERE 
  EXISTS (
    SELECT 1 FROM logins l WHERE l.user_id = u.id AND l.time > NOW() - INTERVAL '1 month'
  )
AND u.status = 'active'
ORDER BY 
  total_payments DESC, average_rating DESC, order_count DESC
LIMIT 10;
