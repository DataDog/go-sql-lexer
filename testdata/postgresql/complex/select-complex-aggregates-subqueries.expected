[
    {
        "expected": "SELECT u.id, u.name, ( SELECT COUNT ( * ) FROM orders o WHERE o.user_id = u.id ), ( SELECT SUM ( amount ) FROM payments p WHERE p.user_id = u.id ), ( SELECT AVG ( rating ) FROM reviews r WHERE r.user_id = u.id ) FROM users u WHERE EXISTS ( SELECT ? FROM logins l WHERE l.user_id = u.id AND l.time > NOW ( ) - INTERVAL ? ) AND u.status = ? ORDER BY total_payments DESC, average_rating DESC, order_count DESC LIMIT ?",
        "statement_metadata": {
            "size": 38,
            "tables": [
                "orders",
                "payments",
                "reviews",
                "users",
                "logins"
            ],
            "commands": [
                "SELECT"
            ],
            "comments": [],
            "procedures": []
        }
    }
]